api/v3/Sqltask.php

```diff
@@ -78,25 +86,50 @@ function _civicrm_api3_sqltask_sort_spec(&$params) {
 function civicrm_api3_sqltask_sort($params) {
 
   try {
+    $tasksorderNew = $params['data'];
+    $taskScreenOrder = $params['task_screen_order'];
+
+    // create new taskorder array for comparison with old taskorder array
+    foreach ($taskScreenOrder as $key => $task) {
+      $task = explode('_', $task);
+      $taskScreenOrder[$key] = $task[0];
+    }
 
-    $tasksOrder = $params['data'];
+    // fetch the task sorting from database
+    $query = "SELECT id FROM civicrm_sqltasks ORDER BY weight;";
+    $result = CRM_Core_DAO::executeQuery($query);
+    $tasksorderDatabase = [];
 
-    foreach($tasksOrder as $key => $task) {
+    while ($result->fetch()) {
+      $tasksorderDatabase[] = $result->id;
+    }
 
+    // create new taskorder array for comparison with old taskorder array
+    foreach ($tasksorderNew as $key => $task) {
       $task = explode('_', $task);
-      $weight = ($key*10) + 10;
+      $tasksorderNew[$key] = $task[0];
+    }
 
+    // check the difference between taskorder array from database and the taskorder array from the screen
+    $checkArray = array_intersect_assoc($taskScreenOrder, $tasksorderDatabase);
+
+    if (count($checkArray) != count($taskScreenOrder)) {
```

Comment:
Since we now only accept exact matches of the order, we don't need `array_intersect_assoc` and this can be simplified to `$checkArray  == $taskScreenOrder`.